lib/intacct_ruby/function.rb

```diff
@@ -98,5 +105,15 @@ def validate_type!
               "one of #{ALLOWED_TYPES}."
       end
     end
+
+    def parse_parameters(parameters)
+      xml_attributes = parameters.is_a?(Hash) \
+                         && parameters.include?(XML_ATTRIBUTES) \
+                         && parameters[XML_ATTRIBUTES].is_a?(Hash) \
+                         && parameters.delete(XML_ATTRIBUTES) \
+                         || nil
+
+      [parameters, xml_attributes]
+    end
```

Comment:
I'm too dumb for that `&& parameters.delete(XML_ATTRIBUTES)`
```suggestion
def parse_parameters(parameters)
  xml_attributes_present = parameters.is_a?(Hash) \
                            && parameters.include?(XML_ATTRIBUTES) \
                            && parameters[XML_ATTRIBUTES].is_a?(Hash) \

  xml_attributes = parameters.delete(XML_ATTRIBUTES) if xml_attributes_present

  [parameters, xml_attributes]
end
```